ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -760,11 +762,21 @@ public void removeListeners(double count) {}
         return null;
       }
       String headerName = HeaderUtil.stripHeaderName(header.getString(0));
-      String headerValue = HeaderUtil.stripHeaderValue(header.getString(1));
+      String headerValue = header.getString(1);
       if (headerName == null || headerValue == null) {
         return null;
       }
-      headersBuilder.add(headerName, headerValue);
+
+      // Use reflection to call addUnsafeNonAscii because it's not available in
+      // older versions of OkHttp that are used internally.
+      // Remove reflection once the internal version of OkHttp is updated to v3.12.0 or higher.
+      try {
+        Method addUnsafeNonAsciiMethod = Headers.Builder.class.getMethod("addUnsafeNonAscii", String.class, String.class);
+        addUnsafeNonAsciiMethod.invoke(headersBuilder, headerName, headerValue);
+      } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+        // Stripping non-ascii characters is needed for the regular `add` method.
+        headersBuilder.add(headerName, HeaderUtil.stripHeaderValue(headerValue));
+      }
```

Comment:
Let's add a couple of variables in the module:

```java
  boolean canUseAddUnsafeAsciiMethod = true;
  Method addUnsafeNonAsciiMethod = null;
```

Then, we can use this code instead:

```suggestion
      if (!canUseAddUnsafeAsciiMethod) {
        // Let's avoid reflection if we know we can't use it.
        headersBuilder.add(headerName, HeaderUtil.stripHeaderValue(headerValue));
      } else {
        try {
          if (this.addUnsafeNonAsciiMethod == null) {
              // Let's try to cache the method here.
              this.addUnsafeNonAsciiMethod = Headers.Builder.class.getMethod("addUnsafeNonAscii", String.class, String.class);
          }
          this.addUnsafeNonAsciiMethod.invoke(headersBuilder, headerName, headerValue);
        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
          // Can't invoke the `addUnsafeNonAsciiMethod`. Since the method does not exists, it is pointless to try with reflection again in the future
          this.canUseAddUnsafeAsciiMethod = false;
          // Stripping non-ascii characters is needed for the regular `add` method.
          headersBuilder.add(headerName, HeaderUtil.stripHeaderValue(headerValue));
        }
      }
```